lib/push_2_tweet_server.js

```diff
@@ -0,0 +1,316 @@
+    p2tHelper = require('./push_2_tweet_helper.js'),
+    hapi = require('hapi'),
+    boom = require('boom'),
+    async = require('async'),
+    Twitter = require('twitter');
+
+var server;
+
+var attendedRequests = 0;
+
+var twitter = new Twitter({
+  'consumer_key': p2tConfig.TWITTER_CONSUMER_KEY,
+  'consumer_secret': p2tConfig.TWITTER_CONSUMER_SECRET,
+  'access_token_key': p2tConfig.TWITTER_ACCESS_TOKEN_KEY,
+  'access_token_secret': p2tConfig.TWITTER_ACCESS_TOKEN_SECRET
+});
+
+/**
+ * Checks that every request sent to the Push 2 Tweet includes the required headers
+ * @param {object} value Headers object
+ * @param {object} options Hapi server header validation configuration object
+ * @param {function} next Hapi server header validation continuation function
+ */
+function validateHeaders(value, options, next) {
+  var error, message;
+
+  var context = {
+    corr: value[p2tConfig.UNICA_CORRELATOR_HEADER] || p2tHelper.getUnicaCorrelator(),
+    trans: p2tHelper.getTransactionId(),
+    op: p2tHelper.getOperationType()
+  };
+
+  attendedRequests++;
+
+  if (!value['fiware-service']) {
+    message = 'error=child "fiware-service" fails because [fiware-service is required]';
+    p2tLogger.warn(
+      context,
+      message
+    );
+    error = boom.badRequest(message);
+    error.output.payload.validation = {source: 'headers', keys: ['fiware-service']};
+    next(error);
+  } else if (!value['fiware-servicepath']) {
+    message = 'child "fiware-servicepath" fails because [fiware-servicepath is required]';
+    p2tLogger.warn(
+      context,
+      message
+    );
+    error = boom.badRequest(message);
+    error.output.payload.validation = {source: 'headers', keys: ['fiware-servicepath']};
+    next(error);
+  }
+  next();
+}
+
+/**
+ * Handler to manage the requests for the version of the component
+ * @param {Object} request The received request
+ * @param {Function} reply The reply function to respond to the requester
+ * @return {*} Returns the version of the Push 2 Tweet component
+ */
+function getVersionHandler(request, reply) {
+  var message = p2tHelper.getVersion();
+  return reply(message);
+}
+
+/**
+ * Returns the logging context associated to a request
+ * @param {Object} request The updateContext request received
+ * @return {Object} The context to be used for logging
+ */
+function getContext(request) {
+  return {
+    corr: request.headers[p2tConfig.UNICA_CORRELATOR_HEADER] ||
+            p2tHelper.getUnicaCorrelator(request),
+    trans: p2tHelper.getTransactionId(),
+    op: p2tHelper.getOperationType(request)
+  };
+}
+
+/**
+ * Attends a received request
+ * @param request The received request
+ * @param callback Callback to be notified when the request has been processed
+ */
+function processOperationRequest(request, callback) {
+  twitter.post(
+    'statuses/update',
+    {
+      status: '(TESTING) Button with id: \'' + request.push2Tweet.operationDescriptor.button + '\'' +
```

Comment:
Maybe it would be a good idea to encapsulate the creation of the Tweet text in a different function, to have it clearly separated in case we want to change it.